source/util.d

```diff
@@ -1,33 +1,131 @@
 module util;
 
+import core.attribute : mustuse;
+import std.algorithm : map;
+import std.array : join;
+import std.format : format;
+import std.meta : All = allSatisfy, IndexOf = staticIndexOf, Map = staticMap;
+import std.range : iota;
 import std.stdio : File, stderr, stdout;
+import std.sumtype;
 
-/**
-Creates an annonymous enum that extends from T and exposes aliases that bind
-to each enum member of T.
+@mustuse struct Result(Success, Error)
+{
+	@safe pure nothrow @nogc:
```

Comment:
For now this is fine, but for types with elaborate construction like copy constructors that throw or do unsafe operations, this won't work well, but I assume this is only for your limited usecase, so I would left it as it is.